sinonjs/samsam
# Experiment type files

This PR is a work in progress to see if we can generate `.d.ts` files from the JSDoc comments.

#### Background

TypeScript users would like to have better support using Sinon and its libraries

#### Solution

1. Write JSDoc for every public function (this benefits everyone)
1. Generate `.d.ts` file(s) from the JSDoc


#### How to verify

I'm not a TypeScript user, so I don't really know what I am doing here

1. Check out this branch
1. `npm ci`
1. `npm build`
1. `npm link`
1. In another project: `npm link @sinonjs/samsam`
1. Try importing `samsam` into your project and see if the type definitions work as expected (I don't know how to do this)
